README.md
user_okf
========

Gallery with locations

ChangeLog
2012-07-20 Peter Pietzschmann  <peter@pietzschmann.de>

	* initial code generated with kickstarter


tca.php
<?php
if (!defined('TYPO3_MODE')) {
	die ('Access denied.');
}

$TCA['user_okf_photo_images'] = array (
	'ctrl' => $TCA['user_okf_photo_images']['ctrl'],
	'interface' => array (
		'showRecordFieldList' => 'hidden,starttime,endtime,fe_group,image_id,cat_id,image_name,image_thumb_file,image_allow_comments,image_allow_itsme,image_comments,image_hits,image_votes'
	),
	'feInterface' => $TCA['user_okf_photo_images']['feInterface'],
	'columns' => array (
		'hidden' => array (		
			'exclude' => 1,
			'label'   => 'LLL:EXT:lang/locallang_general.xml:LGL.hidden',
			'config'  => array (
				'type'    => 'check',
				'default' => '0'
			)
		),
		'starttime' => array (		
			'exclude' => 1,
			'label'   => 'LLL:EXT:lang/locallang_general.xml:LGL.starttime',
			'config'  => array (
				'type'     => 'input',
				'size'     => '8',
				'max'      => '20',
				'eval'     => 'date',
				'default'  => '0',
				'checkbox' => '0'
			)
		),
		'endtime' => array (		
			'exclude' => 1,
			'label'   => 'LLL:EXT:lang/locallang_general.xml:LGL.endtime',
			'config'  => array (
				'type'     => 'input',
				'size'     => '8',
				'max'      => '20',
				'eval'     => 'date',
				'checkbox' => '0',
				'default'  => '0',
				'range'    => array (
					'upper' => mktime(3, 14, 7, 1, 19, 2038),
					'lower' => mktime(0, 0, 0, date('m')-1, date('d'), date('Y'))
				)
			)
		),
		'fe_group' => array (		
			'exclude' => 1,
			'label'   => 'LLL:EXT:lang/locallang_general.xml:LGL.fe_group',
			'config'  => array (
				'type'  => 'select',
				'items' => array (
					array('', 0),
					array('LLL:EXT:lang/locallang_general.xml:LGL.hide_at_login', -1),
					array('LLL:EXT:lang/locallang_general.xml:LGL.any_login', -2),
					array('LLL:EXT:lang/locallang_general.xml:LGL.usergroups', '--div--')
				),
				'foreign_table' => 'fe_groups'
			)
		),
		'image_id' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_images.image_id',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'cat_id' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_images.cat_id',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'image_name' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_images.image_name',		
			'config' => array (
				'type' => 'input',	
				'size' => '30',	
				'eval' => 'required,trim',
			)
		),
		'image_thumb_file' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_images.image_thumb_file',		
			'config' => array (
				'type' => 'input',	
				'size' => '30',	
				'eval' => 'trim',
			)
		),
		'image_allow_comments' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_images.image_allow_comments',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'image_allow_itsme' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_images.image_allow_itsme',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'image_comments' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_images.image_comments',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'image_hits' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_images.image_hits',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'image_votes' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_images.image_votes',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
	),
	'types' => array (
		'0' => array('showitem' => 'hidden;;1;;1-1-1, image_id, cat_id, image_name, image_thumb_file, image_allow_comments, image_allow_itsme, image_comments, image_hits, image_votes')
	),
	'palettes' => array (
		'1' => array('showitem' => 'starttime, endtime, fe_group')
	)
);



$TCA['user_okf_photo_categories'] = array (
	'ctrl' => $TCA['user_okf_photo_categories']['ctrl'],
	'interface' => array (
		'showRecordFieldList' => 'hidden,cat_id,cat_name,cat_description,cat_parent_id,cat_hits,cat_order,cat_hires,cat_since,cat_time,cat_location,cat_type,auth_viewcat,auth_vote,auth_postcomment,auth_votepic'
	),
	'feInterface' => $TCA['user_okf_photo_categories']['feInterface'],
	'columns' => array (
		'hidden' => array (		
			'exclude' => 1,
			'label'   => 'LLL:EXT:lang/locallang_general.xml:LGL.hidden',
			'config'  => array (
				'type'    => 'check',
				'default' => '0'
			)
		),
		'cat_id' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_categories.cat_id',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'cat_name' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_categories.cat_name',		
			'config' => array (
				'type' => 'input',	
				'size' => '30',	
				'eval' => 'required,trim',
			)
		),
		'cat_description' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_categories.cat_description',		
			'config' => array (
				'type' => 'text',
				'cols' => '30',
				'rows' => '5',
				'wizards' => array(
					'_PADDING' => 2,
					'RTE' => array(
						'notNewRecords' => 1,
						'RTEonly'       => 1,
						'type'          => 'script',
						'title'         => 'Full screen Rich Text Editing|Formatteret redigering i hele vinduet',
						'icon'          => 'wizard_rte2.gif',
						'script'        => 'wizard_rte.php',
					),
				),
			)
		),
		'cat_parent_id' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_categories.cat_parent_id',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'cat_hits' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_categories.cat_hits',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'cat_order' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_categories.cat_order',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'cat_hires' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_categories.cat_hires',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'cat_since' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_categories.cat_since',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'cat_time' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_categories.cat_time',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'cat_location' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_categories.cat_location',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'cat_type' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_categories.cat_type',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'auth_viewcat' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_categories.auth_viewcat',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'auth_vote' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_categories.auth_vote',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'auth_postcomment' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_categories.auth_postcomment',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'auth_votepic' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_categories.auth_votepic',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
	),
	'types' => array (
		'0' => array('showitem' => 'hidden;;1;;1-1-1, cat_id, cat_name, cat_description;;;richtext[]:rte_transform[mode=ts_css|imgpath=uploads/tx_userokf/rte/], cat_parent_id, cat_hits, cat_order, cat_hires, cat_since, cat_time, cat_location, cat_type, auth_viewcat, auth_vote, auth_postcomment, auth_votepic')
	),
	'palettes' => array (
		'1' => array('showitem' => '')
	)
);



$TCA['user_okf_locations'] = array (
	'ctrl' => $TCA['user_okf_locations']['ctrl'],
	'interface' => array (
		'showRecordFieldList' => 'hidden,starttime,endtime,id,cat_id,premium,name,typ,typ2,typ3,strasse,plz,ort,email,telefon,fax,url,offen,dresscode,age,beschreibung,hits,since'
	),
	'feInterface' => $TCA['user_okf_locations']['feInterface'],
	'columns' => array (
		'hidden' => array (		
			'exclude' => 1,
			'label'   => 'LLL:EXT:lang/locallang_general.xml:LGL.hidden',
			'config'  => array (
				'type'    => 'check',
				'default' => '0'
			)
		),
		'starttime' => array (		
			'exclude' => 1,
			'label'   => 'LLL:EXT:lang/locallang_general.xml:LGL.starttime',
			'config'  => array (
				'type'     => 'input',
				'size'     => '8',
				'max'      => '20',
				'eval'     => 'date',
				'default'  => '0',
				'checkbox' => '0'
			)
		),
		'endtime' => array (		
			'exclude' => 1,
			'label'   => 'LLL:EXT:lang/locallang_general.xml:LGL.endtime',
			'config'  => array (
				'type'     => 'input',
				'size'     => '8',
				'max'      => '20',
				'eval'     => 'date',
				'checkbox' => '0',
				'default'  => '0',
				'range'    => array (
					'upper' => mktime(3, 14, 7, 1, 19, 2038),
					'lower' => mktime(0, 0, 0, date('m')-1, date('d'), date('Y'))
				)
			)
		),
		'id' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_locations.id',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'cat_id' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_locations.cat_id',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'premium' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_locations.premium',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'name' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_locations.name',		
			'config' => array (
				'type' => 'text',
				'cols' => '30',	
				'rows' => '5',
			)
		),
		'typ' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_locations.typ',		
			'config' => array (
				'type' => 'text',
				'cols' => '30',	
				'rows' => '5',
			)
		),
		'typ2' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_locations.typ2',		
			'config' => array (
				'type' => 'text',
				'cols' => '30',	
				'rows' => '5',
			)
		),
		'typ3' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_locations.typ3',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'strasse' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_locations.strasse',		
			'config' => array (
				'type' => 'text',
				'cols' => '30',	
				'rows' => '5',
			)
		),
		'plz' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_locations.plz',		
			'config' => array (
				'type' => 'input',	
				'size' => '10',	
				'max' => '5',	
				'eval' => 'trim',
			)
		),
		'ort' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_locations.ort',		
			'config' => array (
				'type' => 'text',
				'cols' => '30',	
				'rows' => '5',
			)
		),
		'email' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_locations.email',		
			'config' => array (
				'type' => 'text',
				'cols' => '30',	
				'rows' => '5',
			)
		),
		'telefon' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_locations.telefon',		
			'config' => array (
				'type' => 'text',
				'cols' => '30',	
				'rows' => '5',
			)
		),
		'fax' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_locations.fax',		
			'config' => array (
				'type' => 'text',
				'cols' => '30',	
				'rows' => '5',
			)
		),
		'url' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_locations.url',		
			'config' => array (
				'type' => 'text',
				'cols' => '30',	
				'rows' => '5',
			)
		),
		'offen' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_locations.offen',		
			'config' => array (
				'type' => 'text',
				'cols' => '30',
				'rows' => '5',
				'wizards' => array(
					'_PADDING' => 2,
					'RTE' => array(
						'notNewRecords' => 1,
						'RTEonly'       => 1,
						'type'          => 'script',
						'title'         => 'Full screen Rich Text Editing|Formatteret redigering i hele vinduet',
						'icon'          => 'wizard_rte2.gif',
						'script'        => 'wizard_rte.php',
					),
				),
			)
		),
		'dresscode' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_locations.dresscode',		
			'config' => array (
				'type' => 'text',
				'cols' => '30',	
				'rows' => '5',
			)
		),
		'age' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_locations.age',		
			'config' => array (
				'type' => 'text',
				'cols' => '30',	
				'rows' => '5',
			)
		),
		'beschreibung' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_locations.beschreibung',		
			'config' => array (
				'type' => 'text',
				'cols' => '30',
				'rows' => '5',
				'wizards' => array(
					'_PADDING' => 2,
					'RTE' => array(
						'notNewRecords' => 1,
						'RTEonly'       => 1,
						'type'          => 'script',
						'title'         => 'Full screen Rich Text Editing|Formatteret redigering i hele vinduet',
						'icon'          => 'wizard_rte2.gif',
						'script'        => 'wizard_rte.php',
					),
				),
			)
		),
		'hits' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_locations.hits',		
			'config' => array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'checkbox' => '0',
				'range'    => array (
					'upper' => '1000',
					'lower' => '10'
				),
				'default' => 0
			)
		),
		'since' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_locations.since',		
			'config' => array (
				'type' => 'text',
				'cols' => '30',	
				'rows' => '5',
			)
		),
	),
	'types' => array (
		'0' => array('showitem' => 'hidden;;1;;1-1-1, id, cat_id, premium, name, typ, typ2, typ3, strasse, plz, ort, email, telefon, fax, url, offen;;;richtext[]:rte_transform[mode=ts_css|imgpath=uploads/tx_userokf/rte/], dresscode, age, beschreibung;;;richtext[]:rte_transform[mode=ts_css|imgpath=uploads/tx_userokf/rte/], hits, since')
	),
	'palettes' => array (
		'1' => array('showitem' => 'starttime, endtime')
	)
);
?>

ext_tables.php
<?php
if (!defined('TYPO3_MODE')) {
	die ('Access denied.');
}

t3lib_div::loadTCA('tt_content');
$TCA['tt_content']['types']['list']['subtypes_excludelist'][$_EXTKEY.'_pi1']='layout,select_key,pages';


t3lib_extMgm::addPlugin(array(
	'LLL:EXT:user_okf/locallang_db.xml:tt_content.list_type_pi1',
	$_EXTKEY . '_pi1',
	t3lib_extMgm::extRelPath($_EXTKEY) . 'ext_icon.gif'
),'list_type');


t3lib_extMgm::addToInsertRecords('user_okf_photo_images');

$TCA['user_okf_photo_images'] = array (
	'ctrl' => array (
		'title'     => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_images',		
		'label'     => 'image_name',	
		'tstamp'    => 'tstamp',
		'crdate'    => 'crdate',
		'cruser_id' => 'cruser_id',
		'default_sortby' => 'ORDER BY crdate',	
		'delete' => 'deleted',	
		'enablecolumns' => array (		
			'disabled' => 'hidden',	
			'starttime' => 'starttime',	
			'endtime' => 'endtime',	
			'fe_group' => 'fe_group',
		),
		'dynamicConfigFile' => t3lib_extMgm::extPath($_EXTKEY).'tca.php',
		'iconfile'          => t3lib_extMgm::extRelPath($_EXTKEY).'icon_user_okf_photo_images.gif',
	),
);


t3lib_extMgm::addToInsertRecords('user_okf_photo_categories');

$TCA['user_okf_photo_categories'] = array (
	'ctrl' => array (
		'title'     => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_photo_categories',		
		'label'     => 'cat_name',	
		'tstamp'    => 'tstamp',
		'crdate'    => 'crdate',
		'cruser_id' => 'cruser_id',
		'default_sortby' => 'ORDER BY cat_order',	
		'delete' => 'deleted',	
		'enablecolumns' => array (		
			'disabled' => 'hidden',
		),
		'dynamicConfigFile' => t3lib_extMgm::extPath($_EXTKEY).'tca.php',
		'iconfile'          => t3lib_extMgm::extRelPath($_EXTKEY).'icon_user_okf_photo_categories.gif',
	),
);


t3lib_extMgm::addToInsertRecords('user_okf_locations');

$TCA['user_okf_locations'] = array (
	'ctrl' => array (
		'title'     => 'LLL:EXT:user_okf/locallang_db.xml:user_okf_locations',		
		'label'     => 'uid',	
		'tstamp'    => 'tstamp',
		'crdate'    => 'crdate',
		'cruser_id' => 'cruser_id',
		'default_sortby' => 'ORDER BY crdate',	
		'delete' => 'deleted',	
		'enablecolumns' => array (		
			'disabled' => 'hidden',	
			'starttime' => 'starttime',	
			'endtime' => 'endtime',
		),
		'dynamicConfigFile' => t3lib_extMgm::extPath($_EXTKEY).'tca.php',
		'iconfile'          => t3lib_extMgm::extRelPath($_EXTKEY).'icon_user_okf_locations.gif',
	),
);
?>

ext_localconf.php
<?php
if (!defined('TYPO3_MODE')) {
	die ('Access denied.');
}

t3lib_extMgm::addPItoST43($_EXTKEY, 'pi1/class.user_okf_pi1.php', '_pi1', 'list_type', 1);

t3lib_extMgm::addPageTSConfig('

	# ***************************************************************************************
	# CONFIGURATION of RTE in table "user_okf_photo_categories", field "cat_description"
	# ***************************************************************************************
RTE.config.user_okf_photo_categories.cat_description {
  hidePStyleItems = H1, H4, H5, H6
  proc.exitHTMLparser_db=1
  proc.exitHTMLparser_db {
    keepNonMatchedTags=1
    tags.font.allowedAttribs= color
    tags.font.rmTagIfNoAttrib = 1
    tags.font.nesting = global
  }
}
');
t3lib_extMgm::addPageTSConfig('

	# ***************************************************************************************
	# CONFIGURATION of RTE in table "user_okf_locations", field "offen"
	# ***************************************************************************************
RTE.config.user_okf_locations.offen {
  hidePStyleItems = H1, H4, H5, H6
  proc.exitHTMLparser_db=1
  proc.exitHTMLparser_db {
    keepNonMatchedTags=1
    tags.font.allowedAttribs= color
    tags.font.rmTagIfNoAttrib = 1
    tags.font.nesting = global
  }
}
');
t3lib_extMgm::addPageTSConfig('

	# ***************************************************************************************
	# CONFIGURATION of RTE in table "user_okf_locations", field "beschreibung"
	# ***************************************************************************************
RTE.config.user_okf_locations.beschreibung {
  hidePStyleItems = H1, H4, H5, H6
  proc.exitHTMLparser_db=1
  proc.exitHTMLparser_db {
    keepNonMatchedTags=1
    tags.font.allowedAttribs= color
    tags.font.rmTagIfNoAttrib = 1
    tags.font.nesting = global
  }
}
');
?>

pi1/class.user_okf_pi1.php
<?php
/***************************************************************
*  Copyright notice
*
*  (c) 2012 Peter Pietzschmann <peter@pietzschmann.de>
*  All rights reserved
*
*  This script is part of the TYPO3 project. The TYPO3 project is
*  free software; you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 2 of the License, or
*  (at your option) any later version.
*
*  The GNU General Public License can be found at
*  http://www.gnu.org/copyleft/gpl.html.
*
*  This script is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  This copyright notice MUST APPEAR in all copies of the script!
***************************************************************/
/**
 * [CLASS/FUNCTION INDEX of SCRIPT]
 *
 * Hint: use extdeveval to insert/update function index above.
 */

require_once(PATH_tslib.'class.tslib_pibase.php');


/**
 * Plugin 'OKF' for the 'user_okf' extension.
 *
 * @author	Peter Pietzschmann <peter@pietzschmann.de>
 * @package	TYPO3
 * @subpackage	user_okf
 */
class user_okf_pi1 extends tslib_pibase {
	var $prefixId      = 'user_okf_pi1';		// Same as class name
	var $scriptRelPath = 'pi1/class.user_okf_pi1.php';	// Path to this script relative to the extension dir.
	var $extKey        = 'user_okf';	// The extension key.
	var $pi_checkCHash = true;
	
	/**
	 * The main method of the PlugIn
	 *
	 * @param	string		$content: The PlugIn content
	 * @param	array		$conf: The PlugIn configuration
	 * @return	The content that is displayed on the website
	 */
	function main($content, $conf) {
		$this->conf = $conf;
		$this->pi_setPiVarDefaults();
		$this->pi_loadLL();
		
	
		$content='
			<strong>This is a few paragraphs:</strong><br />
			<p>This is line 1</p>
			<p>This is line 2</p>
	
			<h3>This is a form:</h3>
			<form action="'.$this->pi_getPageLink($GLOBALS['TSFE']->id).'" method="POST">
				<input type="text" name="'.$this->prefixId.'[input_field]" value="'.htmlspecialchars($this->piVars['input_field']).'">
				<input type="submit" name="'.$this->prefixId.'[submit_button]" value="'.htmlspecialchars($this->pi_getLL('submit_button_label')).'">
			</form>
			<br />
			<p>You can click here to '.$this->pi_linkToPage('get to this page again',$GLOBALS['TSFE']->id).'</p>
		';
	
		return $this->pi_wrapInBaseClass($content);
	}
}



if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/user_okf/pi1/class.user_okf_pi1.php'])	{
	include_once($TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/user_okf/pi1/class.user_okf_pi1.php']);
}

?>

ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext "user_okf".
#
# Auto generated 20-07-2012 16:15
#
# Manual updates:
# Only the data in the array - everything else is removed by next
# writing. "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = array(
	'title' => 'OKF',
	'description' => 'OKF',
	'category' => 'plugin',
	'author' => 'Peter Pietzschmann',
	'author_email' => 'peter@pietzschmann.de',
	'shy' => '',
	'dependencies' => '',
	'conflicts' => '',
	'priority' => '',
	'module' => '',
	'state' => 'beta',
	'internal' => '',
	'uploadfolder' => 0,
	'createDirs' => 'uploads/tx_userokf/rte/',
	'modify_tables' => '',
	'clearCacheOnLoad' => 0,
	'lockType' => '',
	'author_company' => '',
	'version' => '0.0.0',
	'constraints' => array(
		'depends' => array(
		),
		'conflicts' => array(
		),
		'suggests' => array(
		),
	),
	'_md5_values_when_last_written' => 'a:15:{s:9:"ChangeLog";s:4:"2913";s:10:"README.txt";s:4:"ee2d";s:12:"ext_icon.gif";s:4:"1bdc";s:17:"ext_localconf.php";s:4:"d19b";s:14:"ext_tables.php";s:4:"8936";s:14:"ext_tables.sql";s:4:"8596";s:27:"icon_user_okf_locations.gif";s:4:"475a";s:34:"icon_user_okf_photo_categories.gif";s:4:"4ad7";s:30:"icon_user_okf_photo_images.gif";s:4:"a25c";s:16:"locallang_db.xml";s:4:"6b15";s:7:"tca.php";s:4:"7820";s:19:"doc/wizard_form.dat";s:4:"4adc";s:20:"doc/wizard_form.html";s:4:"1b74";s:26:"pi1/class.user_okf_pi1.php";s:4:"69bc";s:17:"pi1/locallang.xml";s:4:"d5d6";}',
);

?>